src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action['type']) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          stateCopy[key] = action.extraData[key];
+        }
```

Comment:
you can simply merge this two objects ([more info how](https://frontendinterviewquestions.medium.com/merge-two-objects-in-javascript-40f16605c13c))